src/Nethermind/Nethermind.Consensus/Validators/BlockValidator.cs

```diff
@@ -193,32 +203,70 @@ private bool ValidateWithdrawals(Block block, IReleaseSpec spec, out string? err
         return true;
     }
 
-    private bool ValidateBlobs(Block block, IReleaseSpec spec, out string? error)
+    private bool ValidateTransactions(Block block, IReleaseSpec spec, out int blobsInBlock, out string? error)
```

Comment:
To some extent I like the previous solution better where both validations (IsWellFormed and Blobs) were disjointed. It was easier to read and making 2x for loops is not a problem.

It is also easier to decompose on potential transactions refactoring, that we are considering for optimism.

Reply:
Yes but it is very restrictive in how we handle transactions, me and @jmederosalvarado will be exploring methods to break up transactions into multiple classes.

Reply:
It's still not clear for me how previous version `Validate*` was better for that. But let me rollback this part, cos it make little difference for me, but seems to make your guys life easier.